packages/roc-package-web-app-react/app/server/router.js

```diff
@@ -33,7 +34,10 @@ export default function routes({ createRoutes, createStore, stats, dist }) {
                 this.status = 200;
                 this.body = renderPage();
             } else {
-                const store = createStore ? createStore() : null;
+                const { store, history, url } = setupForRender(createStore, this.url);
+
+                // Give Koa middlewares a chance to interact with the reduxStore
+                // This can be used to dynamically pass some data to the client.
```

Comment:
> Is there no way to get redux-simple-router to dispatch the current location to the store if history already has a location when syncing?    

We might be able to use [these](https://github.com/reactjs/react-router-redux/#pushlocation-replacelocation-gonumber-goback-goforward) to do it but I'm not sure how that works with `basename` that we are now supporting.

> Right, so basically we want to let the eventual onEnter-hooks to get a reference to getState() and maybe even dispatch()?  

I believe that would be nice to have since it would be trivial to do [this](https://github.com/reactjs/react-router/blob/master/examples/auth-flow/app.js#L133).